discovery/db_discovery.go

```diff
@@ -222,20 +217,43 @@ func pmTicketParams(params *net.TicketParams) *pm.TicketParams {
 	}
 }
 
-func (dbo *DBOrchestratorPoolCache) PollOrchestratorInfo(bcast common.Broadcaster) {
+func (dbo *DBOrchestratorPoolCache) PollOrchestratorInfo(ctx context.Context, bcast common.Broadcaster) error {
+
+	if err := dbo.pollOrchestratorInfo(bcast); err != nil {
+		return err
+	}
+
 	ticker := getTicker()
 	go func() {
 		for range ticker.C {
-			orchs, err := dbo.store.SelectOrchs(
-				&common.DBOrchFilter{
-					CurrentRound: dbo.rm.LastInitializedRound(),
-				},
-			)
-
-			if err != nil {
-				glog.Errorf("could not get orchestrators from database: %v", err)
+			select {
```

Comment:
I assume we still want the select statement though but loop indefinitely rather than over the ticker ? 
```
for {
  select {
    case <- ticker.C: 
       // pollOrchestratorInfo
    case <- ctx.Done: 
      // break
}
}
```

Reply:
Oops yep my example code snippet is missing the select statement to wrap those cases.